apps/guardian-ui/src/setup/FederationSetup.tsx

```diff
@@ -47,9 +46,20 @@ export const FederationSetup: React.FC = () => {
     dispatch,
     api,
   } = useSetupContext();
-  const [needsTosAgreement, setNeedsTosAgreement] = useState(!!getEnv().TOS);
+  const [tosConfig, setTosConfig] = useState<{
+    showTos: boolean;
+    tos: string | undefined;
+  }>({ showTos: false, tos: undefined });
```

Comment:
Understood, my question is why are they a single piece of state as an object, instead of:
```ts
const [showTos, setShowTos] = useState(false)
const [tos, setTos] = useState<string>()
```